### PR TITLE
fix(inputs.bind): Convert counters to uint64

### LINKED ISSUE #15866: Input: Bind - strconv.ParseInt : value out of range
telegraf configuration::
```toml
[[inputs.bind]]
urls = ["http://localhost:8053/xml/v3"]
```
error message:
```text
[inputs.bind] Error in plugin: unable to decode XML document: strconv.ParseInt: parsing "18446744073709546041": value out of range
```
telegraf version v1.31.2

related to #6059

### PR DESCRIPTION
**Pull request by srebhan**
## Summary

The bind implementation of the XML v3 endpoint reports counters as unsigned integers through XML. However, the current implementation of the plugin tries to convert those numbers to _signed_ integers which fails for large values of counters leading to a parsing error as shown in the related issue.

This PR changes the internal representation of the counters to unsigned integers for the XML v3 implementation. The PR also adds a `report_counters_as_int` to keep the current, backward-compatible behavior of reporting those counters as _signed_ integer fields. To avoid type conflicts and keep backward-compatibility the default behavior is to continue reporting those counters as integers. 

However, users should change the setting to `false` as soon as possible to avoid parsing errors!

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #15866

### CODE REVIEW

**Comment by rumanmaharjan** [comment]
not able to download package. is it included on v1.32.1 [2024-10-07] ?

**Comment by srebhan** [comment]
@rumanmaharjan I retriggered building of the packages. Please note, those packages will be deleted after 3 days IIRC so it would be good if you could at least download the package before it is removed again...

File: plugins/inputs/bind/bind.go [hunk 1]
```diff
@@ -18,10 +18,11 @@ import (
 var sampleConfig string
 
 type Bind struct {
-	Urls                 []string
-	GatherMemoryContexts bool
-	GatherViews          bool
+	Urls                 []string        `tom:"urls"`
+	GatherMemoryContexts bool            `tom:"gather_memory_contexts"`
+	GatherViews          bool            `tom:"gather_views"`
```
Comment (DStrand1) [inline plugins/inputs/bind/bind.go:23]:
```suggestion
	Urls                 []string        `toml:"urls"`
	GatherMemoryContexts bool            `toml:"gather_memory_contexts"`
	GatherViews          bool            `toml:"gather_views"`
```

**Review by DStrand1 — COMMENTED** [review]
@srebhan Could you also rebase this PR to fix the changelog merge conflict? Thanks!

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**